src/charm.py

```diff
@@ -195,6 +202,20 @@ def __init__(self, *args):
         # Action handlers
         observe(self.on.show_proxied_endpoints_action, self._on_show_proxied_endpoints)
 
+    def _on_cert_transfer_relation_joined(self, event: RelationJoinedEvent):
+        if self.cert.enabled:
```

Comment:
Just `.enabled` may not enough: it only checks if the relation is there and does not look at relation data. The ca cert may not be there yet. I guess we need something like `cert.ready`. Until then may need to also check for `self.cert.ca`.

Reply:
afraid also cert.cert needs to be set, else we'll put a None in the databag and it will break.
A bit unclear to me what the best way to do this is.